xarray/core/variable.py

```diff
@@ -154,9 +160,15 @@ def as_variable(obj: T_DuckArray | Any, name=None) -> Variable | IndexVariable:
             f"explicit list of dimensions: {obj!r}"
         )
 
-    if name is not None and name in obj.dims and obj.ndim == 1:
-        # automatically convert the Variable into an Index
-        obj = obj.to_index_variable()
+    if auto_convert:
+        if name is not None and name in obj.dims and obj.ndim == 1:
+            # automatically convert the Variable into an Index
+            emit_user_level_warning(
```

Comment:
Maybe we should remove this warning since we're not exactly sure yet what the future behavior will be?

Or at least let's make sure this warning won't be emitted in user code unless explicitly calling `as_variable(..., auto_convert=True)`, i.e., in xarray internals all new variables should be converted into index variables if needed explicitly (if needed).

Reply:
Okay, so I tried to ensure that this warning won't be emitted in user code simply by replacing the warning with an exception and running all the tests to see where it raised. The only places it raised were:
- In [`core.dataarray._check_data_shape`](https://github.com/pydata/xarray/blob/7c3d2dd2987f855a44d91993f73a9e1aef0a481d/xarray/core/dataarray.py#L194), which is a place where creating an index is unnecessary, so I fixed it in https://github.com/pydata/xarray/pull/8711/commits/6bbcc8a65daff8c0863d4541c6822f9c3d03b268,
- In the [`TestVariable.test_as_variable`](https://github.com/pydata/xarray/blob/7c3d2dd2987f855a44d91993f73a9e1aef0a481d/xarray/tests/test_variable.py#L1219) test itself, which seems like it actually should raise the warning?

Is that sufficient?

Reply:
Yes that may be enough for now.